fhir-persistence-blob/src/main/java/com/ibm/fhir/persistence/blob/BlobDeletePayload.java

```diff
@@ -44,23 +44,17 @@ public BlobDeletePayload(int resourceTypeId, String logicalId, int version, Stri
      * @throws FHIRPersistenceException
      */
     public Response<Void> run(BlobManagedContainer client) throws FHIRPersistenceException {
-        final StringBuilder blobNameBuilder = new StringBuilder();
-        blobNameBuilder.append(resourceTypeId);
-        blobNameBuilder.append("/");
-        blobNameBuilder.append(logicalId);
-        blobNameBuilder.append("/");
-        blobNameBuilder.append(version);
-        blobNameBuilder.append("/");
-        blobNameBuilder.append(resourcePayloadKey);
-        BlobClient bc = client.getClient().getBlobClient(blobNameBuilder.toString());
-
+        final String blobPath = BlobPayloadSupport.getPayloadPath(resourceTypeId, logicalId, version, resourcePayloadKey);
+        
+        BlobAsyncClient bc = client.getClient().getBlobAsyncClient(blobPath);
         try {
-            Response<Void> response = bc.deleteWithResponse(DeleteSnapshotsOptionType.INCLUDE, null, client.getProperties().getTimeout(), null);
-            return response;
+            // TODO return future so we can avoid waiting for response here
```

Comment:
Is the intent to address this TODO now, or in the future?

Reply:
it's a future enhancement thing, should we need it based on performance results/use cases etc.